activerecord/test/cases/validations/uniqueness_validation_test.rb

```diff
@@ -386,4 +386,21 @@ def test_validate_uniqueness_on_empty_relation
     topic = TopicWithUniqEvent.new
     assert topic.valid?
   end
+
+  def test_does_not_validate_uniqueness_of_if_parent_record_is_validate_false
+    Reply.validates_uniqueness_of(:content)
+
+    Reply.create!(content: "Topic Title")
+
+    reply = Reply.new(content: "Topic Title")
+    reply.save!(validate: false)
+    assert reply.persisted?
+
+    topic = Topic.new(reply_ids: [reply.id])
+    topic.save!
+
+    assert_equal topic.replies.size, 1
+    assert reply.valid?
```

Comment:
@pacoguzman It thought this was odd too. If you look at the new `should_validate?` method:
```
   def should_validate?(record) # :nodoc:
      !record.persisted? || record.changed? || record.marked_for_destruction?
    end
```
When `assert reply.valid?`, the record is [x] persisted, [x] un-changed, [x] not marked for destruction, so the uniqueness validation isn't performed. Before this PR was merged, the validation would have been run, but now it isn't.